models/Request.js

```diff
@@ -0,0 +1,25 @@
+const mongoose = require('mongoose')
+
+const { REQUEST_STATUS } = require('../constants')
+
+const Schema = mongoose.Schema
+
+const RequestSchema = new Schema({
+  itemName: { type: Schema.Types.ObjectId, ref: 'Item' },
+  itemId: { type: Schema.Types.ObjectId, ref: 'Item' },
+  itemOwner: { type: Schema.Types.ObjectId, ref: 'User' },
+  itemOwnerLocation: { type: Schema.Types.ObjectId, ref: 'User' },
+  requestorName: { type: Schema.Types.ObjectId, ref: 'User' },
+  requestorId: { type: Schema.Types.ObjectId, ref: 'User' },
+  requestorEmail: { type: Schema.Types.ObjectId, ref: 'User' },
+  requestorLocation: { type: Schema.Types.ObjectId, ref: 'User' },
+  reqestorNotes: String,
+  daysNeededFor: Number,
+  dateNeededOn: String,
+  status: {
+    type: String,
+    enum: Object.values(REQUEST_STATUS)
+  }
+})
+
+module.exports = mongoose.model('Request', RequestSchema)
```

Comment:
Two things here: 

1. You don't use an Object ref for every field on the other document, just use one per document. So you'd have one for item, one for itemOwner and one for requestor, then you'd use `populate` to fill in the fields. It would be similar to the Item model.
2. I think you'll want to add/update the `pre` middleware in Item and User so that when an Item or a User is deleted the associated requests are deleted too.

Reply:
@phreid I'll leave the middleware update for when an item or user is deleted for another PR so this one doesn't get too big. What do you think of the schema update?

Reply:
I think what you want to do is something like this:

```
const RequestSchema = new Schema({
 item: { type: Schema.Types.ObjectId, ref: 'Item' },}),
 itemOwner: { type: Schema.Types.ObjectId, ref: 'User' },}),
 requestor: { type: Schema.Types.ObjectId, ref: 'User' },}),
 requestorNotes: String,
 daysNeeded: Number,
 [ other request fields ]
```

Then when you call `populate` on a request you get something like:

```
 _id:    , 
 item: {
   __id:
   name:
   type:
   etc. 
 },
 itemOwner: {
   _id:  ,
   username:
   location:  
 },
 requestor: {
   _id:  ,
   username:
   location:  
 }
 etc.
```
That what when an Item or User field changes, you don't need to worry about updating it in all the requests.